### ISSUE TITLE
Bracket matching for code embedded in strings

### ISSUE THREAD

**Issue by MoritzKn** [atom/bracket-matcher#274]
### Description

This package doesn't match brackets if they are inside a string or a comment. ([here is the code](https://github.com/atom/bracket-matcher/blob/master/lib/bracket-matcher-view.coffee#L57))
However in some languages it is possible to embed code inside strings. For example in JavaScript you could write something like this:
```js
`x = ${(15 + pow(6, 2)) * 5}!`
```
But because only brackets outside of strings are matched, this package won't do anything here.

### Steps to Reproduce

1. Open a JS file.
2. Type 
```js
a = `${foo(bar())}`
```
3. Place the cursor on the first bracket.

**Expected behavior:**
The last bracket should be highlighted.

**Actual behavior:**
- No bracket will be highlighted.
- Commands like `go-to-matching-bracket` or `select-inside-brackets` won't work either.

**Reproduces how often:**
100%

### Versions
```
$ atom --version
Atom    : 1.13.0
Electron: 1.3.13
Chrome  : 52.0.2743.82
Node    : 6.5.0
```
```
$ apm --version
apm  1.14.1
npm  3.10.5
node 4.4.5
python 2.7.10
git 2.5.0
```
OS: GNU/Linux (Kubuntu 15.10)

### Additional Information
Perhaps this could be fixed if `BracketMatcherView.isRangeCommentedOrString()` and `BracketMatcherView.isCursorOnCommentOrString()` would return `false` if there is a `.source` scope after the `.string` scope.

**Comment by 50Wliu** [comment]
String and comment matching was intentionally disabled in #253.  I think this is a use case that we should have considered.

**Comment by schlomo** [comment]
Probably the same problem also holds true for shell scripts like this:
```
echo "something $(command arg <(command2))"
```
or 
```
var=$(
  { command ; command ; command ; } | sort
)
```

In both cases I find it annoying that I can't see the matching brackets. IMHO all the brackets and braces in my examples should be matched because they are actually code and not plain strings.

Can you please reconsider your decision and improve #253 or #78? Thanks a lot!

**Comment by MoritzKn** [comment]
I created a PR to fix this: #294
